prow/report/report.go

```diff
@@ -107,22 +111,35 @@ func Report(ghc GithubClient, reportTemplate *template.Template, pj v1.ProwJob)
 	if ghc == nil {
 		return fmt.Errorf("trying to report pj %s, but found empty github client", pj.ObjectMeta.Name)
 	}
+
 	if !pj.Spec.Report {
 		return nil
 	}
+
 	refs := pj.Spec.Refs
-	if len(refs.Pulls) != 1 {
-		return fmt.Errorf("prowjob %s has %d pulls, not 1", pj.ObjectMeta.Name, len(refs.Pulls))
+	if len(refs.Pulls) > 1 {
+		return fmt.Errorf("prowjob %s has %d pulls, thus will skip report", pj.ObjectMeta.Name, len(refs.Pulls))
 	}
+
 	childDescription := fmt.Sprintf("Waiting on: %s", pj.Spec.Context)
 	if err := reportStatus(ghc, pj, childDescription); err != nil {
 		return fmt.Errorf("error setting status: %v", err)
 	}
+
 	// Report manually aborted Jenkins jobs and jobs with invalid pod specs alongside
 	// test successes/failures.
 	if !pj.Complete() {
 		return nil
 	}
+
+	if len(refs.Pulls) == 0 {
+		if pj.Spec.Type == v1.PresubmitJob {
+			// really should not happen...
+			return fmt.Errorf("presubmit job %s doesn't have associated pulls", pj.ObjectMeta.Name)
```

Comment:
What's the point of checking this in the code right here?

Reply:
Is this enforced somewhere else? Just trying to be defensive here.

Reply:
I can't think of any place we create a presubmit job without pulls

Reply:
seems like I can easily do this with `kubectl create -f my-presubmit-prowjob-with-no-refs.yaml`, right?

Reply:
But then we should argue that every part of our codebase is defensive ... /shrug